components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/ApisApiServiceImpl.java

```diff
@@ -3731,6 +3731,9 @@ public Response createNewAPIVersion(String newVersion, String apiId, Boolean def
                 String serviceName = existingAPI.getServiceInfo("name");
                 ServiceCatalogImpl serviceCatalog = new ServiceCatalogImpl();
                 ServiceEntry service = serviceCatalog.getServiceByNameAndVersion(serviceName, serviceVersion, tenantId);
+                if (service == null) {
+                    throw new APIManagementException(ExceptionCodes.from(ExceptionCodes.ERROR_WHILE_TRYING_TO_DISCOVER_SERVICES));
```

Comment:
Its better to use a new ExceptionCode for this.. Above one is not relevant for service catalog. 
Error message can indicate that the service version is not found with status code 404.

ExceptionCodes.from() is used when you want to pass a parameterized error response. If it is not needed you can just use the ExceptionCode as it is. See https://github.com/wso2/carbon-apimgt/wiki/Exception-handling-in-REST-APIs